src/calendar/mod.rs

```diff
@@ -1,71 +1,16 @@
-use crate::clients::google::calendar::*;
-use crate::clients::google::token_storage::TokenStorage;
-use crate::clients::google::GoogleClient;
-use chrono::prelude::*;
-use std::convert::From;
-use std::error;
-use time::Duration;
-
-#[derive(Debug, Clone, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct Event {
-    pub summary: Option<String>,
-    pub location: Option<String>,
-    pub description: Option<String>,
-    pub start_time: Option<String>,
-    pub end_time: Option<String>,
-}
-
-impl From<&CalendarEvent> for Event {
-    fn from(ce: &CalendarEvent) -> Self {
-        Event {
-            summary: ce.summary.clone(),
-            location: ce.location.clone(),
-            description: ce.description.clone(),
-            start_time: match &ce.start {
-                Some(start_time) => start_time.date_time.clone(),
-                None => None,
-            },
-            end_time: match &ce.end {
-                Some(end_time) => end_time.date_time.clone(),
-                None => None,
-            },
-        }
-    }
-}
-
-#[derive(Debug, Clone)]
-pub struct Calendar<T: TokenStorage> {
-    calendars: Vec<String>,
-    google_client: GoogleClient<T>,
-}
-
-impl<T: TokenStorage> Calendar<T> {
-    pub fn new(calendars: Vec<String>, google_client: GoogleClient<T>) -> Self {
-        Self {
-            calendars,
-            google_client,
-        }
-    }
-
-    pub fn get_events(&self) -> Result<Vec<Event>, Box<error::Error>> {
-        let start = dbg!(Utc::now());
-        let end = dbg!(start + Duration::weeks(1));
-
-        let mut all_events: Vec<Event> = Vec::new();
-
-        for calendar in &self.calendars {
-            let mut events: Vec<Event> = self
-                .google_client
-                .get_events(
-                    calendar,
-                    &start.format("%Y-%m-%dT%H:%M:%SZ").to_string(),
-                    &end.format("%Y-%m-%dT%H:%M:%SZ").to_string(),
-                )
-                .map(|result| result.items.iter().map(Event::from).collect())?;
-            all_events.append(&mut events);
-        }
-
-        Ok(all_events)
+mod config;
+mod facade;
+mod google_calendar;
+mod memory_calendar;
+
+pub use config::CalendarConfig;
+pub use facade::CalendarFacade;
+pub use facade::Event;
+
+#[cfg(test)]
+mod test {
+    #[test]
+    fn in_memory_calendar() {
+        // given the in-memory calendar with some events
```

Comment:
👍 test driven development 👍

Reply:
I najlepsze jest to, że nie testuję nawet tego co wykorzystuję produkcyjnie :D